modules/social_features/social_embed/src/Form/EmbedConsentForm.php

```diff
@@ -0,0 +1,132 @@
+   *
+   * @var \Drupal\Core\Cache\CacheBackendInterface
+   */
+  protected CacheBackendInterface $cacheRender;
+
+  /**
+   * Constructs a \Drupal\system\ConfigFormBase object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The factory for configuration objects.
+   * @param \Drupal\Core\Cache\CacheBackendInterface $cacheRender
+   *   The cache services.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, CacheBackendInterface $cacheRender) {
+    parent::__construct($config_factory);
+    $this->cacheRender = $cacheRender;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('cache.render'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'social_embed_admin_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return [
+      static::SETTINGS,
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config(static::SETTINGS);
+
+    // Add an introduction text to explain what can be done here.
+    // Add warning message.
+    $form['warning_message'] = [
+      '#theme' => 'status_messages',
+      '#message_list' => [
+        'warning' => [
+          [
+            '#type' => 'html_tag',
+            '#tag' => 'strong',
+            '#value' => t("If the settings are changed, then once you submit this form, site's cache will be reset."),
+          ],
+        ],
+      ],
+      '#status_headings' => [
+        'warning' => t('Attention!'),
+      ],
+      '#weight' => -1,
+    ];
+
+    $form['description'] = [
+      '#type' => 'html_tag',
+      '#tag' => 'p',
+      '#value' => $this->t('The following setting will allow the users to be able to provide consent before showing the embedded content. The users will also be able to customize options to allow/disallow embed providers in the user settings.'),
+    ];
+
+    $form['consent'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Site wide consent'),
+      '#open' => TRUE,
+    ];
+
+    $form['consent']['settings'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Allow consent'),
+      '#default_value' => $config->get('settings'),
+      '#description' => $this->t('When disabled, all the embedded content will be shown by default.'),
+    ];
+
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    // Retrieve the configuration.
+    $config = $this->configFactory->getEditable(static::SETTINGS);
+    if ($config->get('settings') != ($new_value = $form_state->getValue('settings'))) {
+      $this->cacheRender->invalidateAll();
```

Comment:
So it seems that what we want to do is make sure that we have cache tags in place for all our embedded content. So once embedded content is part of the render array, we need to add our own cache tag.

If we than submit this form, we can clear the cache for that cache tag which will ensure all the content with that cache tag is rebuild.